lib/screens/scanner_screen.dart

```diff
@@ -190,6 +191,15 @@ class _ScannerScreenState extends State<ScannerScreen> {
           }),
         ),
       );
+    } else if (error is StateError && error.toString().contains('INVALID_BACKUP_DATA')) {
+      Navigator.of(context).push(
+        MaterialPageRoute(
+          builder: (context) => WrongMetaMaskWalletForRecoveryScreen(onPress: () {
+            _updateScannerState(ScannerState.scanning);
```

Comment:
these 2 locs happens on many places, should we create a function for them to avoid duplicate?

Reply:
Both are independent right now. We can create new function calling `_resetScannerController() ` and update ScannerState to `scanning`.